app/controllers/events/view_transcript.py

```diff
@@ -0,0 +1,74 @@
+from app.models.course import Course
+from app.models.courseParticipant import CourseParticipant
+from app.models.program import Program
+from app.models.courseInstructor import CourseInstructor
+from app.models.user import User
+from app.models.term import Term
+from app.models.eventParticipant import EventParticipant
+from app.models.event import Event
+from peewee import DoesNotExist
+
+from peewee import *
+
+def getSLCourseTranscript(user):
+    """
+    Returns a service-learning course list that contains a name of the course, term,
+    list of instructors who teach the course, and hours earned for each course that the user took.
+    :user: model object
+    """
+    username = User.get_by_id(user)
+    sLCourseInformation = (CourseParticipant
+    .select(CourseParticipant.course, CourseParticipant.user, fn.SUM(CourseParticipant.hoursEarned).alias("hoursEarned"))
+    .group_by(CourseParticipant.course, CourseParticipant.user)
+    .where(CourseParticipant.user == username))
+
+    courses = [item for item in sLCourseInformation.objects()]

```

Comment:
I don't think this is necessary, is it? You can iterate through sLCourseInformation without it

Reply:
Yes, we don't need it. We removed that loop and it still works and just passed in slCourseInformation in our for loop below it and it works.